SwiftKit/src/main/java/org/swift/swiftkit/SwiftValueWitnessTable.java

```diff
@@ -0,0 +1,227 @@
+
+    /**
+     * Determine the stride of a Swift type given its type metadata, which is
+     * how many bytes are between successive elements of this type within an
+     * array.
+     * <p>
+     * It is >= the size.
+     *
+     * @param typeMetadata the memory segment must point to a Swift metadata
+     */
+    public static long strideOfSwiftType(MemorySegment typeMetadata) {
+        return getSwiftInt(valueWitnessTable(typeMetadata), SwiftValueWitnessTable.$stride$offset);
+    }
+
+
+    /**
+     * Determine the alignment of the given Swift type.
+     *
+     * @param typeMetadata the memory segment must point to a Swift metadata
+     */
+    public static long alignmentOfSwiftType(MemorySegment typeMetadata) {
+        long flags = getSwiftInt(valueWitnessTable(typeMetadata), SwiftValueWitnessTable.$flags$offset);
+        return (flags & 0xFF) + 1;
+    }
+
+    /**
+     * Produce a layout that describes a Swift type based on its
+     * type metadata. The resulting layout is completely opaque to Java, but
+     * has appropriate size/alignment to model the memory associated with a
+     * Swift type.
+     * <p>
+     * In the future, this layout could be extended to provide more detail,
+     * such as the fields of a Swift struct.
+     *
+     * @param typeMetadata the memory segment must point to a Swift metadata
+     */
+    public static MemoryLayout layoutOfSwiftType(MemorySegment typeMetadata) {
+        long size = sizeOfSwiftType(typeMetadata);
+        long stride = strideOfSwiftType(typeMetadata);
+        long padding = stride - size;
+
+        // constructing a zero-length paddingLayout is illegal, so we avoid doing so
+        MemoryLayout[] layouts = padding == 0 ?
+                new MemoryLayout[]{
+                        MemoryLayout.sequenceLayout(size, JAVA_BYTE)
+                                .withByteAlignment(alignmentOfSwiftType(typeMetadata))
+                } :
+                new MemoryLayout[]{
+                        MemoryLayout.sequenceLayout(size, JAVA_BYTE)
+                                .withByteAlignment(alignmentOfSwiftType(typeMetadata)),
+                        MemoryLayout.paddingLayout(stride - size)
+                };
+
+        return MemoryLayout.structLayout(
+                layouts
+        ).withName(SwiftKit.nameOfSwiftType(typeMetadata, true));
+    }
+
+
+    /**
+     * Offset for the "flags" field within the value witness table.
+     */
+    static final long $flags$offset =
+            $LAYOUT.byteOffset(MemoryLayout.PathElement.groupElement("flags"));
+
+    /**
+     * {@snippet lang=C :
+     * ///void(*destroy)(T *object, witness_t *self);
+     * ///
+     * /// Given a valid object of this type, destroy it, leaving it as an
+     * /// invalid object. This is useful when generically destroying
+     * /// an object which has been allocated in-line, such as an array,
+     * /// struct,or tuple element.
+     * FUNCTION_VALUE_WITNESS(destroy,
```

Comment:
FIXME: I think this is the wrong destroy, but was searching around which one we're "actually" calling. The "just the object" seems to actually end up in `_swift_release_dealloc` which seems okey, however with a wrong object value.

Reply:
This is the right one actually :)